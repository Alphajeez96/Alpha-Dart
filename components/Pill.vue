<template>
  <div
    class="holder bg-primary-green text-primary-darkGreen"
    :class="{
      'bg-primary-red text-primary-darkRed': !completed,
    }"
  >
    <span> {{ text }}</span>
  </div>
</template>

<script>
export default {
  name: "Pill",

  props: {
    text: {
      type: String,
      default: "",
    },
    completed: {
      type: Boolean,
      default: false,
    },
  },
};
</script>

<style lang="scss" scoped>
.holder {
  @apply flex items-center justify-center h-7 w-24 text-xs font-medium rounded-full;
}
</style>
