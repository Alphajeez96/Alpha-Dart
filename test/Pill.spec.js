import { shallowMount } from "@vue/test-utils";
import Pill from "~/components/Pill";

describe("Pill", () => {
  it("renders appropriately", () => {
    expect(shallowMount(Pill).html()).toMatchSnapshot();
  });

  it("Renders text", () => {
    const wrapper = shallowMount(Pill, {
      propsData: {
        text: "pill",
      },
    });

    const textSpan = wrapper.find("span");
    expect(textSpan.text()).toEqual("pill");
  });
});
