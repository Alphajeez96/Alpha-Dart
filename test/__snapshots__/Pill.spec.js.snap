// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pill renders appropriately 1`] = `"<div class=\\"holder bg-primary-green text-primary-darkGreen bg-primary-red text-primary-darkRed\\"><span> </span></div>"`;
